src/components/QuranReader/ReadingView/ShareQuranModal.tsx

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable i18next/no-literal-string */
+import React from 'react';
+
+import classNames from 'classnames';
+import useTranslation from 'next-translate/useTranslation';
+
+// Import styles from the original component
+import styles from '@/components/HomePage/ReadingSection/NewCard/ShareQuranModal.module.scss';
+import { ModalSize } from '@/dls/Modal/Content';
+import Modal from '@/dls/Modal/Modal';
+import ShareButtons from '@/dls/ShareButtons';
+import useIsMobile from '@/hooks/useIsMobile';
+import CloseIcon from '@/icons/close.svg';
+import { getFirstTimeReadingGuideNavigationUrl } from '@/utils/navigation';
+import { getBasePath } from '@/utils/url';
+import { getVerseAndChapterNumbersFromKey } from '@/utils/verse';
+
+interface Props {
+  isOpen: boolean;
+  onClose: () => void;
+  verse?: {
+    verseKey: string;
+  };
+}
+
+const ShareQuranModal: React.FC<Props> = ({ isOpen, onClose, verse }) => {
+  const { t } = useTranslation('common');
+  const isMobile = useIsMobile();
```

Comment:
Consider using CSS media queries instead of detecting mobile via JavaScript to apply conditional class names. Media queries are more performant, reliable, and follow best practices for responsive design. They avoid layout shifts, reduce JS complexity, and work better across edge cases like resized desktop windows or tablets. Use JS only when detecting capabilities CSS can’t handle (e.g., touch support).